extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -10,6 +10,15 @@ class CommunityPageSpecialController extends WikiaSpecialPageController {
 	const MODAL_IMAGE_HEIGHT = 700.0;
 	const MODAL_IMAGE_MIN_RATIO = 0.85;
 	const DEFAULT_MODULES_MAX = 3;
+	// order of permissions is consistent with badges hierarchy in Discussions
+	const PERMISSIONS_HIERARCHY = [
```

Comment:
The more I think about it the more it is clear that const is effectively a dictionary and it's not safe to assume order of iteration (we've had that issue with Perl code a while ago when post-update dicts stopped being predictable).
Thus, I propose to make this PERMISSISONS_TO_BADGES and add PERMISSION_HIERARCHY list for the ordering.